code/Snakefile

```diff
@@ -0,0 +1,41 @@
+# Snakefile to run Amdahl's Law for HPC Workflows
+
+cpu_min = 1
+cpu_max = 32
+
+TASKS = range(cpu_min, cpu_max + 1)  # number of processors: start small for testing!
+
+rule plot:
+    input:
+        expand("amdahl_np{task}.json", task=TASKS)
+    output:
+        "plot.log"
+    log:
+        "slurm_plot.log"
+    resources:
+        tasks=1
+    conda:
+        "amdahl"
+    shell:
+        "python3 plot_amdahl.py {input} > {output}"
```

Comment:
Snakemake does have [a specific feature to handle scripts](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#python), so I think may be worth mentioning that. I'm not saying that this rule should be change to use the `script` directive, but rather that users should be aware of that, maybe a callout box would work.

Reply:
Interesting! I'm not sure how `script` interacts with `cluster`, though, and we want the plotting to be done on a compute node.